src/Controller/LoginController.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+class LoginController {
+
+    const COOKIE_EXPIRE_TIME = 3600;
+
+    /**
+     *
+     * @var mysqli
+     */
+    private $connection;
+
+    /**
+     *
+     * @var PasswordHasher
+     */
+    private $passwordHasher;
+
+    /**
+     *
+     * @var User
+     */
+    private $user;
```

Comment:
Pavieniai entity kaip klasės laukai naudojami labai retai (tuo tarpu pvz array of entity - pasitaiko neretai). Taip yra todėl kad visą klasės instance pririši prie vieno entity. Šiuo atveju prie vieno vartotojo pvz mokinio Jono. Jeigu turėtum kažkada naudotą dalyko pažymių vidurkio skaičiuotuvą ir norint kiekvienam mokiniui sužinoti vidurkį reiktų kurti po naują skaičiuotuvo instance'ą kas yra neoptimalu. 
Šiuo atveju tas user yra prisijungęs/dabartinis vartotojas ir jis bus tik vienas (t.y. vienas vartotojas niekada neprijungs 2 user), tai entity saugojimas klasės lauke nėra ypač nelogiškas variantas, bet pilnai būtų galima apsieiti ir user entity, kur reikia, perduodant kaip funkcijų argumentą. Palieku tau spręsti ar nori perdaryti